Src/FluentAssertions/Xml/Equivalency/XmlReaderValidator.cs

```diff
@@ -209,5 +221,17 @@ private Failure ValidateText()
 
             return null;
         }
+
+        private Failure ValidateAgainstNulls()
+        {
+            if ((expectationIterator is null) ^ (subjectIterator is null))
```

Comment:
When trying this out with
```cs
var document = null as XDocument;
var otherDocument = XDocument.Parse("<parent><child /></parent>");
document.Should().BeEquivalentTo(otherDocument);
```

I get a somewhat unhelpful failure message.
```
Expected document to be equivalent to <null>, but found "".
```

Instead of adding checks in `XmlReaderValidator`, I would prefer if we null checked at call-site.
It's more lines of code, but it avoids adding complexity to `XmlReaderValidator`.

Something like this
```cs
public AndConstraint<XDocumentAssertions> BeEquivalentTo(XDocument expected, string because = "", params object[] becauseArgs)
{
    if (Subject is null && expected is null)
    {
        return new AndConstraint<XDocumentAssertions>(this);
    }

    bool success = Execute.Assertion
        .BecauseOf(because, becauseArgs)
        .ForCondition(Subject is not null && expected is not null)
        .FailWith(() => new FailReason("Expected {context:subject} to be equivalent to {0}{reason}, but found {1}.",
            Subject?.ToString(), expected?.ToString()));

    if (success)
    {
        using (XmlReader subjectReader = Subject.CreateReader())
        using (XmlReader otherReader = expected.CreateReader())
        {
            var xmlReaderValidator = new XmlReaderValidator(subjectReader, otherReader, because, becauseArgs);
            xmlReaderValidator.Validate(shouldBeEquivalent: true);
        }
    }

    return new AndConstraint<XDocumentAssertions>(this);
}
```

The code above gives this failure message.
```
Expected document to be equivalent to <null>, but found "<parent>
      <child />
    </parent>".
```

Reply:
Why would `XmlReaderValidator` not handle null values on its own? I took this code from `StringEqualityValidator` (`StringValidator`) where it is responsible for the null checks.

Reply:
As `StringValidator` also does it that way I'm fine with it and your solution solves the problem 👍 
It's internal so we can always improve later if we want to.

Reply:
The correct content is now also part of the error message.